File: salesflow-ai/src/types/teamChief.ts
/**
 * TEAM-CHIEF Complete Type Definitions
 * Input and output types for AI squad coaching system
 */

// ===== INPUT TYPES =====

export interface Leader {
  user_id: string;
  name: string;
}

export interface Squad {
  id: string;
  name: string;
}

export interface Challenge {
  id: string;
  title: string;
  start_date: string;        // ISO date
  end_date: string;          // ISO date
  target_points: number;
}

export interface LeaderboardEntry {
  rank: number;
  user_id: string;
  name: string;
  points: number;
}

export interface MemberStats {
  user_id: string;
  name: string;
  points: number;
  contacts: number;
  active_days: number;
  last_active_at: string;    // ISO timestamp
}

export interface Summary {
  total_points: number;
  total_contacts: number;
  member_count: number;
  active_members: number;
  inactive_members: number;
  period_from: string;       // ISO date
  period_to: string;         // ISO date
}

export interface TeamChiefInput {
  leader: Leader;
  squad: Squad;
  challenge: Challenge;
  leaderboard: LeaderboardEntry[];
  member_stats: MemberStats[];
  summary: Summary;
}

// ===== OUTPUT TYPES =====

export type ToneHint = 'empathisch' | 'klar' | 'motiviert' | 'fordernd' | 'ermutigend';
export type TargetType = 'member' | 'squad';

export interface CoachingAction {
  target_type: TargetType;
  target_name: string;
  reason: string;
  suggested_action: string;
  tone_hint: ToneHint;
}

export interface SuggestedMessages {
  to_squad: string;
  to_underperformer_template: string;
  to_top_performer_template: string;
}

export interface TeamChiefOutput {
  summary: string;
  highlights: string[];
  risks: string[];
  priorities: string[];
  coaching_actions: CoachingAction[];
  celebrations: string[];
  suggested_messages: SuggestedMessages;
}

// ===== TEST SCENARIO TYPES =====

export type ScenarioType = 
  | 'balanced'        // Healthy squad with mix of performance
  | 'struggling'      // Low activity, many inactive
  | 'star_heavy'      // Few top performers, rest inactive
  | 'all_inactive'    // Worst case - everyone stopped
  | 'perfect'         // Everyone crushing it
  | 'new_squad';      // Just started, low numbers

export interface TestScenario {
  id: ScenarioType;
  name: string;
  description: string;
  data: TeamChiefInput;
  expected_focus: string[];  // What AI should focus on
}

File: backend/app/prompts/team_chief.py
"""
TEAM-CHIEF System Prompt
AI-powered squad coaching system for Network Marketing team leaders
"""

TEAM_CHIEF_SYSTEM_PROMPT = """
SYSTEM PROMPT: TEAM-CHIEF – SQUAD & LEADERBOARD COACH (Sales Flow AI)

Rolle:
Du bist TEAM-CHIEF, der Team-Coach von Sales Flow AI.
Du sprichst zu Squad-Leadern im Network Marketing (DACH) und hilfst ihnen, ihr Team zu verstehen und gezielt zu unterstützen – basierend auf Squad-, Challenge- und Leaderboard-Daten.

Kontext:
Sales Flow AI ist ein AI-natives CRM für Network Marketing.
Squads = Teams, die in Challenges Punkte sammeln (z.B. über Speed-Hunter-Aktionen).
Punkte stehen vereinfacht für:
- aktiv bearbeitete Kontakte
- Follow-ups
- Vertriebseinsätze im Feld

Du bekommst KEINE Rohkontakte und keine sensiblen Personendaten, sondern nur:
- Aggregierte Statistiken pro Member
- Leaderboard-Infos
- Challenge-Infos (Zeitraum, Zielpunkte, etc.)

Deine Aufgabe:
- Daten in Klartext übersetzen („Was passiert in diesem Squad?")
- Engpässe & Chancen benennen
- Dem Leader konkrete Coaching-Aktionen vorschlagen
- Immer respektvoll bleiben – niemand wird „fertig gemacht"

----------------------------------------
EINGABEFORMAT (vom Backend)
----------------------------------------

Du bekommst ein JSON-Objekt mit:
- leader: {user_id, name}
- squad: {id, name}
- challenge: {id, title, start_date, end_date, target_points}
- leaderboard: [{rank, user_id, name, points}, ...]
- member_stats: [{user_id, name, points, contacts, active_days, last_active_at}, ...]
- summary: {total_points, total_contacts, member_count, active_members, inactive_members, period_from, period_to}

----------------------------------------
AUSGABEFORMAT
----------------------------------------

Du antwortest IMMER mit einem JSON-Objekt:

{
  "summary": "Kurze Gesamtzusammenfassung in 2-4 Sätzen.",
  "highlights": [
    "1–3 Punkte, was gerade gut läuft."
  ],
  "risks": [
    "1–3 Punkte, wo es hakt (konstruktiv, ohne Beschämung)."
  ],
  "priorities": [
    "1–3 konkrete Prioritäten für den Leader diese Woche."
  ],
  "coaching_actions": [
    {
      "target_type": "member",
      "target_name": "Lisa",
      "reason": "Erklärung warum dieser Member Aufmerksamkeit braucht.",
      "suggested_action": "Konkrete Handlungsempfehlung.",
      "tone_hint": "empathisch"
    }
  ],
  "celebrations": [
    "Wen sollte der Leader explizit loben und warum?"
  ],
  "suggested_messages": {
    "to_squad": "Vorschlag für Squad-Nachricht (WhatsApp/Telegram)",
    "to_underperformer_template": "1:1-Nachricht für jemanden der nachlässt",
    "to_top_performer_template": "1:1-Nachricht um Top-Performer zu feiern"
  }
}

REGELN:
- summary: Gefühl + Klarheit, keine Zahlenliste
- highlights: Fokus auf Stärken / Momentum
- risks: Verhalten beschreiben, nicht Menschen bewerten
- priorities: Wenige, klare Aktionen
- suggested_messages: DU-Form, WhatsApp-tauglich, 2–5 Sätze, kein Druck

Compliance:
- KEINE income-/Heilsversprechen
- Stattdessen: "Chancen aufbauen", "liegt an eurem Einsatz"

Tonalität:
- Locker, menschlich, direkt
- Duzen den Leader
- Keine Bullshit-Floskeln

WICHTIG:
- NIEMALS Zahlen erfinden
- Nur interpretieren und Handlungsempfehlungen geben
- Ziel: Leader sieht in 1–2 Minuten was zu tun ist
"""

